utils/preprocess.py

```diff
@@ -35,6 +36,32 @@ def __getitem__(self, idx) -> tuple[torch.Tensor, torch.Tensor, str]:
         image_name = self.image_names[idx]
 
         return sand_dust_image, ground_truth_image, image_name
+    
+
+class UnpairedDataset(Dataset):
+    # load all images at once
+    # we use float32 for the image data type
+    def __init__(self, sand_dust_images: List[Image.Image], image_names: List[str]):
+        self.sand_dust_images = sand_dust_images
+        self.image_names = image_names
+        self.w_threshold = 440
+        self.h_threshold = 330
+
+    def __len__(self):
+        return len(self.sand_dust_images)
+
+    def preprocess(self, image: Image) -> torch.Tensor:
+        image = resize_image(image, self.w_threshold, self.h_threshold)
+        image = np.asarray(image) / 255.0
+        image = np.transpose(image, axes=[2, 0, 1]).astype('float32')  # to C, H, W
+        image = torch.from_numpy(image).float()
+        return image
+
+    def __getitem__(self, idx) -> tuple[torch.Tensor, torch.Tensor, str]:
+        sand_dust_image = self.preprocess(self.sand_dust_images[idx])
+        image_name = self.image_names[idx]
+        return sand_dust_image, image_name
```

Comment:
Is the content of this class exactly the same as the code in the PairedDataset class? If so, I think we can create ImageDataset that is for both paired and unpaired datasets (we might need to change variables names in the class though)